llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -1887,6 +1887,17 @@ OpenMPIRBuilder::createTask(const LocationDescription &Loc,
       SplitBlockAndInsertIfThenElse(IfCondition, IfTerminator, &ThenTI,
                                     &ElseTI);
       Builder.SetInsertPoint(ElseTI);
+
+      if (Dependencies.size()) {
+        Function *TaskWaitFn =
+            getOrCreateRuntimeFunctionPtr(OMPRTL___kmpc_omp_taskwait_deps_51);
+        Builder.CreateCall(
```

Comment:
Consider updating the comment starting with
```
In the presence of the `if` clause, the following IR is generated:
```

Reply:
Done